source/actions3/actions/AMDetectorSetDarkCurrentTimeAction.cpp

```diff
@@ -0,0 +1,41 @@
+#include "AMDetectorSetDarkCurrentTimeAction.h"
+#include "beamline/AMBeamlineSupport.h"
+
+AMDetectorSetDarkCurrentTimeAction::AMDetectorSetDarkCurrentTimeAction(AMDetectorSetDarkCurrentTimeActionInfo *info, AMDetector *detector, QObject *parent) :
+    AMAction3(info, parent)
+{
+	detector_ = 0;
+	setDetector(detector);
+}
+
+AMDetectorSetDarkCurrentTimeAction::AMDetectorSetDarkCurrentTimeAction(const AMDetectorSetDarkCurrentTimeAction &other) :
+	AMAction3(other)
+{
+	detector_ = 0;
+	setDetector(other.detector_);
+}
+
+AMDetectorSetDarkCurrentTimeAction::~AMDetectorSetDarkCurrentTimeAction()
+{
+
+}
+
+void AMDetectorSetDarkCurrentTimeAction::startImplementation()
+{
+	if (!detector_)
```

Comment:
This check will cause the action to hang.  Since this is very undesirable, you likely need to call ```setFailed()```.